### PR TITLE
Normalize directory separators before calling LoadLibrary

### LINKED ISSUE #22094: Path with forward-slashes passed to LoadUnmanagedDllFromPath cause load to fail
**Describe the bug**
When a native binary in a package is defined in the `.nuspec` using *forward-slashes*, `AssemblyLoadContext.LoadUnmanagedDllFromPath()` will fail loading it.

Sample `<file>` causing this failure:

**Foo.nuspec**
```xml
<files>
  <file src="nativeassets\*.dll" target="runtimes/win-x64/native" />
</files>
```

**To Reproduce**
1. Define a NuGet package that include a native binary using the above pattern.
2. Load an assembly in an ALC that depend on the native binary and execute some code in it, e.g. using P/Invoke, and resolve specifically to the path in the NuGet cache.

**Expected behavior**
The native binary successfully load.

**Actual behavior**
The native binary fail to load.

**Screenshots**
image

**Additional context**
Spent quite some time figuring out what this problem was about, and what especially fooled me was that `File.Exist(unmanagedPath)` return **true**, because that API clearly deal with mixed slashes. Then when you pass the exact same path on to `AssemblyLoadContext.LoadUnmanagedDllFromPath(path)`, it will raise the above exception.

Tested only on

```
.NET Core SDK (reflecting any global.json):
 Version:   2.2.101
 Commit:    236713b0b7

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.2.101\

Host (useful for support):
  Version: 2.2.0
  Commit:  1249f08fed
```

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #22094

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs [hunk 1]
```diff
@@ -351,6 +351,8 @@ protected IntPtr LoadUnmanagedDllFromPath(string unmanagedDllPath)
             {
                 throw new ArgumentException(SR.Argument_AbsolutePathRequired, nameof(unmanagedDllPath));
             }
+            
+            unmanagedDllPath = PathInternal.RemoveRelativeSegments(unmanagedDllPath, PathInternal.GetRootLength(unmanagedDllPath));
```
Comment (AaronRobinsonMSFT) [inline src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs]: I really don't understand this function call. The issue seems to be related to un-normalized path delimiters. Shouldn't there be a place deeper in the loader that handles that?

Comment (jkoritzinsky) [reply src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs]: The rest of the paths already normalize. I'll see if I can move this lower in the stack.

Comment (jkoritzinsky) [reply src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs]: Moved it down the stack.

**Comment by jkotas** [comment]
Do we need test for it somewhere?

**Pull request merged**